content/markdown/01-basic-r/01-05-packages.rmd

```diff
@@ -0,0 +1,76 @@
+---
+output:
+  pdf_document: default
+  html_document: default
+---
+# Packages in R
+**Author:** Cole Brookson
+**Date:** 22 June 2022
+
+## What Are Packages?
+
+R packages are collections of related functions, code, and data, which can be downloaded and used by any R user, as a way of avoiding having to write all the code to do a task manually themselves. All official packages are submitted and verified through CRAN (Comprehensive R Archive Network). While a number of functions and capabilities ship natively with "base" R, packages typically are sets of functions that perform related tasks (e.g. data management). For example, if in an R script or `.Rmd` file, we begin to type the function `subset()`, we get the following: 
+
+![](./img/code-completion.png)
+We see here that R is suggesting some options to us of functions that start with `sub`. The word in the curly braces, `{base}` indicates that that function is native to "base" R, or the base set of functions and methods that we have access to without loading in other packages. When you download R, a set of packages are imported automatically, including `base`, `stats`, and some others. Packages not downloaded automatically with R need to be explicitly loaded separately. 
+
+There are many many packages, the vast majority of which we will not even talk about. In fact, as of November 2020, more than 16,000 packages were available on CRAN (and that's not including the non-CRAN packages that are available too!).
+
+Note that many packages are termed *dependencies*, that is, they are not the packages we are going to use directly, but they are required on our computers, for the package we **do** want to use to work properly. 
+
+## Loading Packages
+
+We can check what packages we have installed on our computer by running the `library()` command. 
+```{r}
+library()
+```
+On this computer, that returns a large number of packges, but on your computer it may return less. 
```

Comment:
What is in the PDF here is a warning that there are no packages, it doesn't look like a list.